Tournament_Calculator.py

```diff
@@ -0,0 +1,323 @@
+
+
+
+
+#Swiss Pairings Logic
+def generate_pairings(players, previous_pairings=None, results=None):
+    pairings = []
+    if previous_pairings is None:
+        previous_pairings = set()  # Initialize an empty set for previous pairings
+    if results is None:
+        results = {}  # Initialize an empty dictionary for results
+    
+    # Group players based on their current results
+    result_groups = {}
+    for player in players:
+        result = results.get(player, 0)
+        if result not in result_groups:
+            result_groups[result] = []
+        result_groups[result].append(player)
+
+
+    
+    # Sort result groups based on results
+    sorted_result_groups = sorted(result_groups.items(), key=lambda x: x[0], reverse=True)
+
+
+    # Pair players with similar results
+    for i, (result, group) in enumerate(sorted_result_groups):
+        if len(group) % 2 != 0:
+            # If there's an odd number of players in the group, try to pair with the highest-ranked player in the next lowest group
+            next_group_index = i + 1
+            while next_group_index < len(sorted_result_groups):
+                next_result, next_group = sorted_result_groups[next_group_index]
+                if len(next_group) > 0:
+                    odd_player = group[0]
+                    random.shuffle(next_group)
+                    pair = (group[0], next_group[0])
+                    while pair in previous_pairings or pair[::-1] in previous_pairings:

```

Comment:
This is overall pretty complex and hard to read, recommend adding more comments above the if and while blocks to explain what each is doing briefly (the comments you already have are a good start). Another option that's more self-documenting would be to break it into multiple named functions.